### PR TITLE
Update sbt to 1.2.8

### PR DESCRIPTION
**Pull request by regadas**

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -158,15 +158,18 @@ val noBinaryCompatCheck = Set[String]("benchmark", "caliper", "generic", "spark"
 
 def previousVersion(subProj: String) =
   Some(subProj)
-    .filterNot(noBinaryCompatCheck.contains(_))
-    .map { s => "com.twitter" %% ("algebird-" + s) % "0.13.4" }
+    .filterNot(noBinaryCompatCheck.contains)
+    .map { s => "com.twitter" %% ("algebird-" + s) % "0.13.5" }
```
Comment (regadas) [inline build.sbt:162]: with `mima < 0.4.0` some rules where missed and nwe ones were [introduced](https://users.scala-lang.org/t/announcing-mima-0-4-0/4799). `0.13.4` and `0.13.5` might not be binary compatible.

File: project/plugins.sbt [hunk 2]
```diff
@@ -4,13 +4,13 @@ resolvers ++= Seq(
...
 
-addSbtPlugin("com.47deg"          % "sbt-microsites"  % "0.7.0")
-addSbtPlugin("com.eed3si9n"       % "sbt-unidoc"      % "0.4.1")
-addSbtPlugin("com.github.gseitz"  % "sbt-release"     % "1.0.8")
-addSbtPlugin("com.jsuereth"       % "sbt-pgp"         % "1.1.1")
+addSbtPlugin("com.47deg"          % "sbt-microsites"  % "0.9.4")
+addSbtPlugin("com.eed3si9n"       % "sbt-unidoc"      % "0.4.2")
+addSbtPlugin("com.github.gseitz"  % "sbt-release"     % "1.0.11")
+addSbtPlugin("com.jsuereth"       % "sbt-pgp"         % "2.0.0")
 addSbtPlugin("com.lucidchart"     % "sbt-scalafmt"    % "1.15")
```
Comment (regadas) [inline project/plugins.sbt:11]: I think this should be replaced with:
```suggestion
addSbtPlugin("org.scalameta" % "sbt-scalafmt" % "2.0.5")
```

I didn't updated it because it will likely change the current format and it should be on a separate PR.

**Review by nevillelyh — APPROVED** [review]
LGTM @johnynek or anyone else wants to have a look?

**Comment by johnynek** [comment]
This looks great, thanks!

**Comment by johnynek** [comment]
We should set up scala steward for this repo.

**Comment by johnynek** [comment]
added to scala-steward: https://github.com/fthomas/scala-steward/pull/962

**Comment by regadas** [comment]
awesome! 🎉
